src/servers/src/repeated_field.rs

```diff
@@ -414,12 +414,13 @@ impl<'a, T> IntoIterator for &'a mut RepeatedField<T> {
     }
 }
 
-impl<'a, T> IntoIterator for RepeatedField<T> {
+impl<T> IntoIterator for RepeatedField<T> {
     type Item = T;
     type IntoIter = vec::IntoIter<T>;
 
-    fn into_iter(self) -> vec::IntoIter<T> {
-        self.into_iter()
+    fn into_iter(mut self) -> vec::IntoIter<T> {
+        self.vec.truncate(self.len);
```

Comment:
What is this for, Is there any lint or test that reminds you of it?

Reply:
Clippy complains there are two "into_iter" that can be confusing. Retain the one here.